beancount_interpolate/common.py

```diff
@@ -26,6 +26,24 @@ def extract_mark_posting(posting, config):
     return False
 
 
+def get_number_of_txn(begin_date, duration, step, max_txn=float('inf')):
+    """
+    Computes the number of transactions within a given interval and step length.
+
+    Note: As the current implementation is a bit costly with a very large duration
+    and a small step size there is a `max_txn` parameter which can be set.
+    This is meant to be used with the config `max_new_txn`.
+    Upon exceeding `max_txn` the function returns `max_txn + 1`.
+    """
+    n_txn = 0
+    end_date = begin_date + duration
+    while begin_date + n_txn * step < end_date:
+        n_txn += 1
+        if n_txn > max_txn:
+            break
+    return n_txn
```

Comment:
Yup, it's ~50% costlier overall for my ledger.

I timed my ledger 3 times (`$ time bean-check --no-cache ledger/main.bean`):
- master: 16.5-17.0 seconds
- PR: 23.2 - 23.9 seconds

I'm not familiar with `relativedelta`, but does it have a equilavent function/method for manipulating durations and expressing then in various units, like for example [momentjs has with `asMonths` and similar methods](https://momentjs.com/docs/#/durations/)?

Reply:
see [my comment](https://github.com/Akuukis/beancount_interpolate/pull/19#issuecomment-1073299033)